abdiff/core/create_final_records.py

```diff
@@ -83,16 +80,32 @@ def create_final_records(
 
 
 def get_final_records_iter(
-    diffs_dataset: ds.Dataset,
-    metrics_dataset: ds.Dataset,
+    diffs_dataset_path: str,
+    metrics_dataset_path: str,
     metrics_timdex_field_columns: list[str],
 ) -> Generator[pa.RecordBatch, None, None]:
 
     with duckdb.connect(":memory:") as conn:
 
-        # register datasets in DuckDB for use
-        conn.register("diffs", diffs_dataset.to_table())
-        conn.register("metrics", metrics_dataset.to_table())
```

Comment:
Or was it this line that was loading the full `diffs` and `metrics` datasets into memory? 🤔

Reply:
Correct!   The `to.table()` was 100% to blame here.